### PR TITLE
docs(artifacts): Document inflating artifacts from CI builds

### PR DESCRIPTION
**Pull request by ezimanyi**
This will evolve as we get feedback from early users, but this is a first pass at documenting the new ability to extract artifacts from CI builds.

### CODE REVIEW

**Review by dorbin — APPROVED** [review]
@ezimanyi 
Where can I find the Spinnaker-supplied templates?

**Comment by ezimanyi** [comment]
@dorbin : The Spinnaker-supplied templates (of which there is currently only one) are in this directory: https://github.com/spinnaker/echo/tree/master/echo-pipelinetriggers/src/main/resources

That being said, this makes me realize it might make sense to have a list of these included in the documentation.  Do you think that would be a useful addition to this page?

File: reference/artifacts/from-build-triggers.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+When an external CI system triggers a pipeline, Spinnaker can use the CI build information to
+inject relevant artifacts into the pipeline. The instructions here assume that you have
+[set up a CI system](/setup/ci/) and are familiar with using
+[property files](/guides/user/pipeline/expressions/#property-files) to pass variables from
+CI builds to Spinnaker pipelines.
+
+The extraction of artifacts from the build information is done via a Jinja template; the
+template uses the trigger as context and outputs a list of artifacts to inject into the
+pipeline. Spinnaker provides a set of standard templates to use for artifact extraction,
```
Comment (dorbin) [inline reference/artifacts/from-build-triggers.md]: Link from "Spinnaker provides a set of standard templates" to new section at the end of the doc:

## Supplied templates

**Comment by dorbin** [comment]
@ezimanyi 
Aye. :+1:  See my inline comment.

File: reference/artifacts/from-build-triggers.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+[set up a CI system](/setup/ci/) and are familiar with using
+[property files](/guides/user/pipeline/expressions/#property-files) to pass variables from
+CI builds to Spinnaker pipelines.
+
+The extraction of artifacts from the build information is done via a Jinja template; the
+template uses the trigger as context and outputs a list of artifacts to inject into the
+pipeline. Spinnaker provides a set of [standard templates](#supplied-templates) to use for
+artifact extraction, which users can augment with custom templates.
+
+## Selecting a template
```
Comment (dorbin) [inline reference/artifacts/from-build-triggers.md]: s/Selecting/Select/

File: reference/artifacts/from-build-triggers.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    templatePath: <path to the template>
+```
+You can then use the custom template by exporting the following as properties from your
+CI build:
+```sh
+messageFormat=<name of template>
+customFormat=true
+```
+
+## Binding variables into templates
```
Comment (dorbin) [inline reference/artifacts/from-build-triggers.md]: s/Binding/Bind/

File: reference/artifacts/from-build-triggers.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+We could then generate a useful artifact by having the CI job export the following:
+```sh
+group=test.group
+artifact=test-artifact
+version=123
+messageFormat=custom-jar
+customFormat=true
+```
+
+## Supplied templates
```
Comment (dorbin) [inline reference/artifacts/from-build-triggers.md:79]: Add link to dir containing the one template we have so far.

**Review by dorbin — APPROVED** [review]

**Pull request merged**